dali/test/python/test_external_source_pytorch_gpu.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# it is enough to just import all functions from test_internals_operator_external_source
+# nose will query for the methods available and will run them
+# the test_internals_operator_external_source is 99% the same for cupy and numpy tests
+# so it is better to store everything in one file and just call `use_cupy` to switch between the default numpy and cupy
+
+from test_external_source_impl import *
+use_torch(True)
+
+# extra tests, GPU-specific
+import torch
+from test_utils import check_output
+import nvidia.dali.fn as fn
+from nvidia.dali.pipeline import Pipeline
+
+def test_external_source_callback_torch_stream():
+    with torch.cuda.stream(torch.cuda.Stream()):
+        for attempt in range(10):
+            t0 = torch.tensor([attempt * 100 + 1.5], dtype=torch.float32).cuda()
+            increment = torch.tensor([10], dtype=torch.float32).cuda()
+            pipe = Pipeline(1, 3, 0)
+
+            def gen_batch():
+                nonlocal t0
```

Comment:
Why do you need non-local for t0 and not increment?

Reply:
Because I need the same tensor to change - simply returning a new one in PyTorch resulted in synchronization and the error could not be reproduced even when streams were wrong.